text/0486-deprecate-mouseenter.md

```diff
@@ -0,0 +1,137 @@
+
+This works fine in general, but `mouseenter`/`mouseleave` events are special as they do
+not bubble. In the past it still worked nevertheless as jQuery transparently handled this
+for us, as it had special support for event delegation for these events, essentially by using 
+(bubbling) `mouseover` events to replicate the semantics of `mouseenter`/`mouseleave` events.
+
+When support for jQuery-less apps was introduced, this [left a hole](https://github.com/emberjs/ember.js/issues/16591)
+in the jQuery-less EventDispatcher implementation. But as support for those events was and
+still is part of Ember's pubic API, we had no chance other than to [implement support
+for jQuery-less apps](https://github.com/emberjs/ember.js/pull/16603) using the same 
+`mouseover` based approach.
+
+This however comes with a cost: besides an [unresolved issue](https://github.com/emberjs/ember.js/issues/17228)
+the implementation has some performance drawbacks, as it has to process every `mouseover` event on 
+*any* element, create fake `mouseenter`/`mouseleave` events and try to dispatch them, even when 
+not a single component/action needs them.  
+
+Deprecating support for `mousemove` is also proposed, which is a (bubbling) event that does not have the higher 
+implementation cost as `mouseenter`/`mouseleave`, but nevertheless requires the EventDispatcher to optimistically handle
+these extremely high-volume events.
+
+While efforts to make this more "pay as you go" are [possible](https://github.com/emberjs/ember.js/pull/17911), 
+the trade-off of keeping support around still seems unfavorable, as these events fire so
+frequently, while they are (most certainly) very rarely used.
+
+This is even more so given that Glimmer Components with their outerHTML semantics do not 
+work with event handler methods, and `{{action}}` will eventually fade away in favor of
+`{{on}}` using native `addListener()`.
+
+## Transition Path
+
+When a `mouseenter`/`mouseleave`/`mousemove` event is dispatched by the EventDispatcher to a component
+or `{{action}}` listening to that event type, a deprecation warning will be issued.
+
+The linked migration guide will cover these use cases:
+
+#### Ember.Component
+
+Before:
+
+```js
+import Component from '@ember/component';
+
+export default Component.extend({
+  mouseEnter(e) {
+    // do something
+  }
+});
+```
+
+After:
+
+```js
+import Component from '@ember/component';
+import { bind } from '@ember/runloop';
+
+export default Component.extend({
+  mouseenter(e) {
+    // do something
+  },
+  didInsertElement() {
+    this._super(...arguments);
+    this._mouseenter = bind(this, this.mouseenter);
+    this.element.addEventListener('mouseenter', this._mouseenter);
+  },
+  willDestroyElement() {
+    this._super(...arguments);
+    this.element.removeEventListener('mouseenter', this._mouseenter);
+  }
+});
+```
+
+#### `{{action}}` modifier
+
+Before:
+
+```hbs
+<button {{action "handleMouseEnter" on="mouseenter"}}>Hover</button>
```

Comment:
Is this correct? I thought we used `mouseEnter` for the `on` value here also?

Reply:
Right! 🤦‍♂️ Fixed.